lib/kb_ObjectInfo/CreateAssemblyReport.py

```diff
@@ -0,0 +1,139 @@
+        self.dfu = DataFileUtil(self.callback_url)
+
+               
+    def assembly_overview(self,obj_list):
+        header = "OVERVIEW"
+        this_list = [["Assembly Name","Type","Assembly Type"]]
+        
+        for assembly in obj_list['data']:
+            name = "Assembly Data Object"
+            type_assem = ''
+            object_type = ''
+            if 'info' in assembly:
+                name = assembly['info'][1]
+                object_type = assembly['info'][2]
+            if 'type' in assembly['data']:
+                type_assem = assembly['data']['type']
+
+            this_list.extend([[name,object_type,type_assem]])
+        
+        return (header,this_list)
+        
+    def assembly_metadata(self,obj_list):
+        header = "METADATA"
+        this_list = [['Assembly Name','Assembly ID', 'DNA size', 'GC content', 'Number contigs',
+                'FastA handle reference', 'MD5', 'Type', 'Taxon reference','Original filename']]
+                
+        list = ['assembly_id', 'dna_size', 'gc_content', 'num_contigs',
+                'fasta_handle_ref', 'md5', 'type', 'taxon_ref']
+                
+        for assembly in obj_list['data']:
+            name = "Assembly Data Object"
+            if 'info' in assembly:
+                name = assembly['info'][1]
+
+            # Create the row for the one assembly
+            assem_list = [name]
+            for item in list:
+                if item in assembly['data']:
+                    assem_list.append(str(assembly['data'][item]))
+                else:
+                    assem_list.append(" ")
+
+            if 'fasta_handle_info' in assembly['data'] and 'node_file_name' in assembly['data']['fasta_handle_info']:
+                assem_list.append(assembly['data']['fasta_handle_info']['node_file_name'])
+            else:
+                assem_list.append(" ")
+                
+            # Add the row to the list that will be returned
+            this_list.extend([assem_list])
+            
+        return (header,this_list)
+        
+    def assembly_dnabases(self,obj_list):
+        header = "DNA Composition"
+        this_list = [["Assembly Name","Total DNA Bases",
+                        "A Count","A Percent","C Count","C Percent",
+                        "G Count","G Percent","T Count","T Percent"]]
+ 
+        for assembly in obj_list['data']:
+            name = "Assembly Data Object"
+            if 'info' in assembly:
+                name = assembly['info'][1]
+            dna_size = 1.0
+            pct = 1.00
+            if 'dna_size' in assembly['data']:
+                dna_size = assembly['data']['dna_size']
+            
+            assem_list = [name,str(dna_size)]
+            for base in ["A","C","G","T"]:
+                pct = round(100 * assembly['data']['base_counts'][base] / dna_size,2)
+                assem_list.append(str(assembly['data']['base_counts'][base]))
```

Comment:
Is there a reason to convert this (and the line below) to a string?
If so, perhaps a comment line to explain it?

Reply:
okay, though it is weird that the csv module insists on strings, since csv deals with numbers just fine. The reason I mention it is that a string in csv can always be written with quotes, while a number shouldn't be. That is, you can write a line like
foo, bar, 123
as
"foo", "bar", 123
and they are equivalent, since csv parsers will treat non-numeric data as strings.
But if you do
"foo", "bar", "123"
which is what a csv output could look like if 123 is a string, then it is no longer equivalent, because a parser will treat "123" as a string, not a number, and the code using it will need to convert it.

Then again, when consumed by a spreadsheet program, I'll bet they each have different approaches to dealing with this.